src/textual_fspicker/parts/directory_navigation.py

```diff
@@ -459,6 +464,12 @@ def _on_option_list_option_highlighted(
             assert isinstance(event.option, DirectoryEntry)
             self.post_message(self.Highlighted(self, event.option.location))
 
+    def on_click(self, event: events.Click) -> None:
+        self.last_event_doubleclick = event.chain >= 2
```

Comment:
`last_event_doubleclick` is an internal detail so should be named with a leading underscore. Also, it would be a good idea to introduce a default state for it in `__init__`.

Also, as a small nit-pick: I wonder if it should have a more appropriate name? It's also being set in `on_key`, which obviously has nothing to do with clicking (in a mouse sense). I'm not sure off the top of my head what a better name might be, but perhaps we can tweak that?